server.js

```diff
@@ -27,7 +19,54 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+const PER_PAGE = 10
+
+app.get('/books', (req, res) => {
+  const { page } = req.query
+  const startIndex = PER_PAGE * +page
+  let booksData = books
+
+  const { lang } = req.query
+  const { title } = req.query
+  const { rating } = req.query
+
+
+  if (rating) {
+    if (rating === 'asc') {
+      booksData.sort(function (a, b) { return a.average_rating - b.average_rating })
+    } else if (rating === 'desc') {
+      booksData.sort(function (a, b) { return b.average_rating - a.average_rating })
+    }
+  } else {
+    booksData.sort(function (a, b) { return a.bookID - b.bookID })
+  }
+
+  if (lang) {
+    booksData = booksData.filter(book => book.language_code === lang)
+  }
+
+  if (title) {
+    booksData = booksData.filter(book => book.title.toString().toLowerCase().includes(title.toLowerCase()))
+  }
+  const length = booksData.length
```

Comment:
There's no real issue with this, but I don't think this variable makes the code any more readable - just makes the line count longer. I'd just use `booksData.length` in the totalPages line further down.

Reply:
I have this since it is possible to query by language or title and the size of the array would change after that.
If I use booksData.length after the slice the length would be incorrect and if I use books.length the number of pages could be wrong if there was any query filtering the data.
Any suggestions?